kolibri/plugins/facility_management/assets/src/views/UserCreatePage.vue

```diff
@@ -0,0 +1,295 @@
+            label: this.coreString('adminLabel'),
+            value: UserKinds.ADMIN,
+          },
+        ];
+      },
+    },
+    mounted() {
+      this.$store.dispatch('notLoading');
+    },
+    methods: {
+      goToUserManagementPage(onComplete) {
+        this.$router.push(this.$router.getRoute('USER_MGMT_PAGE'), onComplete);
+      },
+      usernameIsUnique(value) {
+        return !this.facilityUsers.find(
+          ({ username }) => username.toLowerCase() === value.toLowerCase()
+        );
+      },
+      submitForm() {
+        this.formSubmitted = true;
+        if (!this.formIsValid) {
+          return this.focusOnInvalidField();
+        }
+        this.busy = true;
+        this.$store
+          .dispatch('userManagement/createFacilityUser', {
+            username: this.username,
+            full_name: this.fullName,
+            id_number: this.idNumber,
+            gender: this.gender,
+            birth_year: this.birthYear,
+            role: {
+              kind: this.newUserRole,
+              collection: this.currentFacilityId,
+            },
+            password: this.password,
+          })
+          .then(() => {
+            this.handleSubmitSuccess();
+          })
+          .catch(error => {
+            this.handleSubmitFailure(error);
+          });
+      },
+      handleSubmitSuccess() {
+        this.goToUserManagementPage(() => {
+          this.$store.dispatch(
+            'createSnackbar',
+            this.$tr('userCreatedNotification', { username: this.username })
+          );
+        });
+      },
+      handleSubmitFailure(error) {
+        this.caughtErrors = CatchErrors(error, [ERROR_CONSTANTS.USERNAME_ALREADY_EXISTS]);
+        this.busy = false;
+        if (this.caughtErrors.length > 0) {
+          this.focusOnInvalidField();
+        } else {
+          this.$store.dispatch('handleApiError', error);
+        }
+      },
+      focusOnInvalidField() {
+        this.$nextTick().then(() => {
+          if (!this.fullNameValid) {
+            this.$refs.fullNameTextbox.focus();
+          } else if (!this.usernameValid) {
+            this.$refs.usernameTextbox.focus();
+          } else if (!this.passwordValid) {
+            this.$refs.passwordTextbox.focus();
+          }
+        });
+      },
+    },
+    $trs: {
+      createNewUserHeader: 'Create new user',
+      classCoachDescription: "Can only instruct classes that they're assigned to",
+      facilityCoachDescription: 'Can instruct all classes in your facility',
+      userCreatedNotification: "User account for '{username}' was created",
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .coach-selector {
+    padding: 0;
+    margin: 0;
+    border: 0;
+  }
+
+  .select {
+    margin: 18px 0 36px;
+  }
+
+  .buttons {
+    button:first-of-type {
+      margin-left: 0;
+    }
+  }
+
+  .narrow-container {
+    width: 500px;
+    margin: auto;
+  }
+
+  .form {
+    max-width: 400px;
+    min-height: 950px;
```

Comment:
I noticed that the container was quite a bit larger than the actual form. What's the reason for this?

Reply:
Some of the Keen-UI form components have a fixed width of 400px, we make the group the same width for consistency. I think later we'll try to extend the KPageContainer to fit to contents when we need it to.

Reply:
But the `min-height`?

Reply:
That's to make sure everything fits when _everything_ is in the form. The form is the tallest when the User Kind dropdown + Radio button fieldset is there (try editing a user account that's not your own). Without the extra height, the birth year dropdown will overflow the bottom.

Reply:
Seems like just a `padding-bottom` could be set to always give the bottom buffer needed for the dropdown?

Reply:
yeah, i can do that

Reply:
it does make the container jump a bit when you toggle the kind from coach to non-coach.